### PR TITLE
Improve naming of methods to add arguments to git calls

### PR DESCRIPTION
**Pull request by delvh**
| Old name                            | Intention                                | Problem                                                                                           | New name                                      |
| ------------------------------------ | -------------------------------------- | ------------------------------------------------------------------------------------- | -------------------------------------- | 
| `AddDynamicArguments` | add untrusted arguments | what's the meaning of "dynamic"?                                             | `AddUntrustedArguments`  |
| `AddArguments`                 | add trusted arguments     | from the name, it is unclear if you can/should pass untrusted arguments | `AddTrustedArguments`      |

With this, we can ensure that all commands use the correct method to prevent security issues:
String literals should be added using `AddTrustedArguments`, the rest using `AddUntrustedArguments`.

The only remaining question is what is better: The old name for `AddTrustedArguments` as it is significantly shorter and hence stands out more in comparison to `AddUntrustedArguments`, or the new name which clarifies its purpose better?

### CODE REVIEW

File: modules/gitgraph/graph_test.go [hunk 1]
```diff
@@ -269,446 +269,446 @@ func TestCommitStringParsing(t *testing.T) {
 	}
 }
 
-var testglyphs = `*
```
Comment (yardenshoham) [inline modules/gitgraph/graph_test.go]: Is this change intended?

Comment (delvh) [reply modules/gitgraph/graph_test.go]: Argh! I always make the same mistake with that file.
The problem is, I ensured I replaced everything using
```sh
find . -type f -name '*.go' -print -exec vim {} -c ':%s/$OLD_FUNCTION/$NEW_FUNCTION/g' -c ':wq' \;
```
The problem is, my vim is configured to throw out trailing whitespace.
Will revert it now.

Comment (zeripath) [reply modules/gitgraph/graph_test.go]: vscode F2 is generally a safer way to refactor function names (unless of course they're called dynamically)

**Review by zeripath — APPROVED** [review]

**Review by yardenshoham — CHANGES_REQUESTED** [review]
Remove `modules/private/.restore_repo.go.swp`

**Review by yardenshoham — APPROVED** [review]
LGTM

File: modules/git/command.go [hunk 2]
```diff
@@ -103,18 +103,19 @@ func (c *Command) SetDescription(desc string) *Command {
 	return c
 }
 
-// AddArguments adds new git argument(s) to the command. Each argument must be safe to be trusted.
-// User-provided arguments should be passed to AddDynamicArguments instead.
-func (c *Command) AddArguments(args ...CmdArg) *Command {
+// AddTrustedArguments adds new git argument(s) to the command.
+// Each argument must be safe to be trusted, so only string literals should be passed.
+// User-provided arguments should be passed to AddUntrustedArguments instead.
+func (c *Command) AddTrustedArguments(args ...CmdArg) *Command {
```
Comment (wxiaoguang) [inline modules/git/command.go:109]: 1. That's not true: `so only string literals should be passed.`
2. Why I didn't choose `AddTrustedArguments`: it could be misleading. For example, there are codes like:
    * `cmd.AddTrustedArguments("-c", CmdArg("user.name="+opts.Committer.Name)`: is the `CommiterName` trusted or not?
    * `cmd.AddTrustedArguments(CmdArg("--prefix=" + filepath.Base(strings.TrimSuffix(repo.Path, ".git")) + "/"))`: is the generated path trusted?

**Comment by wxiaoguang** [comment]
To answer the question in the table:

> what's the meaning of "dynamic"?

`dynamic` means the argument is constructed dynamically.

> from the name, it is unclear if you can/should pass untrusted arguments

It's clear that it only accepts literal string or `CmdArg`.

Even after this PR's change, `AddTrustedArguments` could be still misleading, and even worse, it's dangerous to accept `AddTrustedArguments(CmdArg(userInputDynamicInput))`, then Remote-Code-Execution occurs. The `Trusted` word in the function name will make reviewers ignore the danger of `userInputDynamicInput`

**Comment by wxiaoguang** [comment]
ps: In mind mind at that  time, I was trying to name the argument type by `CmdArgTrusted`, then `AddArguments(CmdArgTrusted(userDynamicInput))` could be more clear to reviewers.

Although I didn't take  it at that time (due to the long-name), however now,  maybe it seems good to have `CmdArgTrusted`, it's definitely clearer than `AddTrustedArguments`.

File: modules/git/command.go (hunk 2 above)
Comment (zeripath) [reply modules/git/command.go:109]: but I think in these calls you're declaring that you believe that `"user.name="+opts.Committer.Name` is a safe and trustworthy argument that does not need further checking by git.Command.

Comment (wxiaoguang) [reply modules/git/command.go:109]: 2 is  just my worry about the readability. 

`Trusted` is a very strong word, it could lead to misunderstand more easliy.

And, using `AddTrustedArguments` could also have other problems (the comment below).

**Comment by wxiaoguang** [comment]
My plan was
* #22678

Now here it is.

**Comment by 6543** [comment]
replaced by #22678
